src/RestApi/Controllers/Products.php

```diff
@@ -219,7 +219,7 @@ protected function get_product_data( $product, $context = 'view' ) {
 			'has_options'    => $product->has_options(),
 			'is_purchasable' => $product->is_purchasable(),
 			'is_in_stock'    => $product->is_in_stock(),
-			'add_to_cart'    => [
+			'add_to_cart'    => (object) [
```

Comment:
The schema for these doesn't define `array` as the type, but `object`. Therefore, when validating against schema, these were not matching up. When you get the data from a REST Request you don't get back JSON.

Also, when you decode JSON you can explicitly choose whether or not you want a assoc array or objects, so I'm not sure the validation would work if we did some fancy encoding/decoding.

So we have a choice I think; use objects for things defined in schema as objects, or allow validation to pass for arrays even if they are defined as objects.

Reply:
The schema defines what is returned in a _response_ to the client (when it has been converted to a json object) not how it is shaped in the php.  It's defined as `object` because that's how it is in the response.  The difficulty you're hitting here is that you're testing the php response before it has been converted to a json object and returned back to the client.

I'm really reluctant to be doing premature casting to objects in the schema when it's going to be cast to an object anyways on a REST request. It's also against convention. Are there other examples where you've seen REST API schema definitions being cast to an object?

I would think if you're wanting to validate this, you should validate what the client sees, which means running the result of the internal request (which is an array) through`wp_json_encode` before validating.

Reply:
Nvm what I wrote above, for some reason I was thinking this was the schema being affected but it's clearly not and my argument is completely silly 🤦‍♂